### ISSUE TITLE
Fail to package pytorch-transformer model as spaCy package

### ISSUE THREAD

**Issue by fredriko** [explosion/spacy-pytorch-transformers#53]
**TL;DR**

I fail to create a spaCy package of a model that I'm able to load with HuggingFace's pytorch-transformers. The error message I get when trying to load the packaged model indicates that I'm missing a tokenizer (final row of error message below).

How do I go about properly package a pytorch-transformer model as a spaCy package?

```
FileNotFoundError: [Errno 2] No such file or directory: '/Users/fredriko/venv/spacy-rubert/lib/python3.6/site-packages/ru_pytt_rubert_cased-1.0.0-py3.6.egg/ru_pytt_rubert_cased/ru_pytt_rubert_cased-1.0.0/tokenizer'
```

**VERBOSE**

The model I want to access in spaCy is RuBERT (http://docs.deeppavlov.ai/en/master/features/models/bert.html).

* I've downloaded the Tensorflow model, 
* converted it into a pytorch equivalent using the functionality provided by HuggingFace. 
* I'm able to load it in pytorch-transformers, and have it produce embeddings for a sample text. 
* I saved it from pytorch-transformers and, 
* used the `python -m spacy package` command to convert it to a spaCy package (with a meta.json file similar to the ones accompanying existing spaCy BERT models), followed by `python3 setup.py install`.
* The presence of the package was confirmed with `pip freeze | grep rubert`

When I try to load the newly created package like so:

```
import spacy

nlp = spacy.load("ru_pytt_rubert_cased")
```
I get the following stacktrace:

```
/Users/fredriko/venv/spacy-rubert/bin/python /Users/fredriko/PycharmProjects/rubert/spacy_example.py
Traceback (most recent call last):
  File "/Users/fredriko/PycharmProjects/rubert/spacy_example.py", line 11, in <module>
    nlp = spacy.load("ru_pytt_rubert_cased")
  File "/Users/fredriko/venv/spacy-rubert/lib/python3.6/site-packages/spacy/__init__.py", line 27, in load
    return util.load_model(name, **overrides)
  File "/Users/fredriko/venv/spacy-rubert/lib/python3.6/site-packages/spacy/util.py", line 134, in load_model
    return load_model_from_package(name, **overrides)
  File "/Users/fredriko/venv/spacy-rubert/lib/python3.6/site-packages/spacy/util.py", line 155, in load_model_from_package
    return cls.load(**overrides)
  File "/Users/fredriko/venv/spacy-rubert/lib/python3.6/site-packages/ru_pytt_rubert_cased-1.0.0-py3.6.egg/ru_pytt_rubert_cased/__init__.py", line 12, in load
    return load_model_from_init_py(__file__, **overrides)
  File "/Users/fredriko/venv/spacy-rubert/lib/python3.6/site-packages/spacy/util.py", line 196, in load_model_from_init_py
    return load_model_from_path(data_path, meta, **overrides)
  File "/Users/fredriko/venv/spacy-rubert/lib/python3.6/site-packages/spacy/util.py", line 179, in load_model_from_path
    return nlp.from_disk(model_path)
  File "/Users/fredriko/venv/spacy-rubert/lib/python3.6/site-packages/spacy/language.py", line 836, in from_disk
    util.from_disk(path, deserializers, exclude)
  File "/Users/fredriko/venv/spacy-rubert/lib/python3.6/site-packages/spacy/util.py", line 636, in from_disk
    reader(path / key)
  File "/Users/fredriko/venv/spacy-rubert/lib/python3.6/site-packages/spacy/language.py", line 823, in <lambda>
    p, exclude=["vocab"]
  File "tokenizer.pyx", line 389, in spacy.tokenizer.Tokenizer.from_disk
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/pathlib.py", line 1181, in open
    opener=self._opener)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/pathlib.py", line 1035, in _opener
    return self._accessor.open(self, flags, mode)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/pathlib.py", line 387, in wrapped
    return strfunc(str(pathobj), *args)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/fredriko/venv/spacy-rubert/lib/python3.6/site-packages/ru_pytt_rubert_cased-1.0.0-py3.6.egg/ru_pytt_rubert_cased/ru_pytt_rubert_cased-1.0.0/tokenizer'
```

I'm using

```
spacy==2.1.8
spacy-pytorch-transformers==0.3.0
```
The `meta.json` file looks like this:

```
{
  "lang":"ru",
  "lang_factory":"pytt",
  "requirements":[
    "spacy-pytorch-transformers>=0.3.0"
  ],
  "name":"pytt_rubert_cased",
  "version":"1.0.0",
  "spacy_version":">=2.1.8",
  "description":"This is a Pytorch version of the Keras/Tensorflow Russian BERT model (RuBERT) trained and made available by DeepPavlov",
  "notes":"Requires the `spacy-pytorch-transformers` package to be installed. A CUDA-compatible GPU is advised for reasonable performance.",
  "author":"DeepPavlov",
  "email":"",
  "url":"https://github.com/explosion/spacy-pytorch-transformers",
  "license":"MIT",
  "pipeline":[
    "sentencizer",
    "pytt_wordpiecer",
    "pytt_tok2vec"
  ],
  "sources":[
    "rubert_cased_L-12_H-768_A-12_v1"
  ]
}

```

**Comment by ines** [comment]
> used the python -m spacy package command to convert it to a spaCy package (with a meta.json file similar to the ones accompanying existing spaCy BERT models), followed by python3 setup.py install.

Ah, that's not going to work! It still needs to be a **serialized spaCy model**, so you'll need to load your RuBERT weights with `spacy-pytorch-transformers` and then call `nlp.to_disk` to save it out as a spaCy model. The rest should then work as expected.

See here for how to set up the pipeline and get an `nlp` object: https://github.com/explosion/spacy-pytorch-transformers#setting-up-the-pipeline

**Comment by fredriko** [comment]
Magic. It works! Wonderful :)

**Comment by fredriko** [comment]
For the sake of completeness, I've outlined the steps required to convert a BERT Tensorflow model to a spaCy package here: https://github.com/fredriko/bert-tensorflow-pytorch-spacy-conversion
